### ISSUE TITLE
feat: add API Token to the accessible tenants APIs

### ISSUE THREAD

**Issue by OmarIthawi** [nelc/futurex-openedx-extensions#35]
To make it callable only from within the UI, so the UI to fully control this API.

This will make adding Captcha effective to improve the security of the API.

**Comment by shadinaif** [comment]
@tehreem-sadat please remember that this is a breaking change, we must coordinate with the front-end deployment for it

**Comment by OmarIthawi** [comment]
yes please, we'll need to coordinate with the frontend engineer on this issue.
